Xamarin.Forms.DataGrid/Utils/ReflectionUtils.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Reflection;
+
+namespace Xamarin.Forms.DataGrid.Utils
+{
+    public static class ReflectionUtils
+    {
+        private const char IndexBeginOp = '[';
+        private const char IndexEndOp = ']';
+        private const char PropertyOfOp = '.';
+
+        public static object GetValueByPath(object obj, string path)
```

Comment:
There is no root observation in this code that needs to be added. 

i.e: If I set the `PropertyName ` to `Student.Lectures[0].Teacher.Name` and change the `Teacher`, context will not updates.